server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -7844,9 +7844,10 @@ public UserVm restoreVirtualMachine(final Account caller, final long vmId, final
             }
             vm.setPassword(password);
         }
+        UserVmDetailVO userVmPasswordDetail = userVmDetailsDao.findDetail(vm.getId(), VmDetailConstants.PASSWORD);
         if (needRestart) {
             try {
-                if (vm.getDetail(VmDetailConstants.PASSWORD) != null) {
+                if (Objects.nonNull(userVmPasswordDetail)) {
```

Comment:
@Pearl1594 
does this work ?
```
                if (password != null) {
```

Reply:
I believe that should work too. I'll check and update for simplicity.